yb-voyager/cmd/live_migration_integration_test.go

```diff
@@ -1686,3 +1686,1332 @@ $$ LANGUAGE plpgsql;`,
+				decimal_small
+			) VALUES
+			(
+				555555555.555555555,
+				-777.777,
+				0.000,
+				888.888888888888888,
+				100.500000,
+				75.50
+			);`,
+
+			// DECIMAL UPDATE #1: Update DECIMAL values
+			`UPDATE test_schema.decimal_edge_cases
+			SET decimal_large = 999999999.999999999,
+			    decimal_negative = -1000.001
+			WHERE id = 1;`,
+
+			// DECIMAL UPDATE #2: Update with repeating decimals and trailing zeros
+			`UPDATE test_schema.decimal_edge_cases
+			SET decimal_high_precision = 0.123456789012345,
+			    decimal_scientific = 3.1415926000
+			WHERE id = 2;`,
+
+			// DECIMAL DELETE #1: Test DECIMAL row deletion
+			`DELETE FROM test_schema.decimal_edge_cases WHERE id = 3;`,
+		},
+		CleanupSQL: []string{
+			`DROP SCHEMA IF EXISTS test_schema CASCADE;`,
+		},
+	})
+
+	defer lm.Cleanup()
+
+	t.Log("=== Setting up containers ===")
+	err := lm.SetupContainers(context.Background())
+	testutils.FatalIfError(t, err, "failed to setup containers")
+
+	t.Log("=== Setting up schema ===")
+	err = lm.SetupSchema()
+	testutils.FatalIfError(t, err, "failed to setup schema")
+
+	t.Log("=== Starting export data ===")
+	err = lm.StartExportData(true, nil)
+	testutils.FatalIfError(t, err, "failed to start export data")
+
+	t.Log("=== Starting import data ===")
+	err = lm.StartImportData(true, map[string]string{
+		"--log-level": "debug",
+	})
+	testutils.FatalIfError(t, err, "failed to start import data")
+
+	t.Log("=== Waiting for snapshot complete (10 datatypes × 3 rows = 30 rows) ===")
+	err = lm.WaitForSnapshotComplete(map[string]int64{
+		`test_schema."string_edge_cases"`:         3, // 3 rows with STRING edge cases
+		`test_schema."json_edge_cases"`:           3, // 3 rows with JSON edge cases
+		`test_schema."enum_edge_cases"`:           3, // 3 rows with ENUM edge cases
+		`test_schema."bytes_edge_cases"`:          3, // 3 rows with BYTES edge cases
+		`test_schema."datetime_edge_cases"`:       3, // 3 rows with DATETIME edge cases
+		`test_schema."uuid_ltree_edge_cases"`:     3, // 3 rows with UUID/LTREE edge cases
+		`test_schema."map_edge_cases"`:            3, // 3 rows with MAP edge cases
+		`test_schema."interval_edge_cases"`:       3, // 3 rows with INTERVAL edge cases
+		`test_schema."zonedtimestamp_edge_cases"`: 3, // 3 rows with ZONEDTIMESTAMP edge cases
+		`test_schema."decimal_edge_cases"`:        3, // 3 rows with DECIMAL edge cases
+	}, 120) // 1 minute should be plenty for small rows
+	testutils.FatalIfError(t, err, "failed to wait for snapshot complete")
+
+	t.Log("=== Validating snapshot data ===")
+	err = lm.ValidateDataConsistency([]string{`test_schema."string_edge_cases"`, `test_schema."json_edge_cases"`, `test_schema."enum_edge_cases"`, `test_schema."bytes_edge_cases"`, `test_schema."datetime_edge_cases"`, `test_schema."uuid_ltree_edge_cases"`, `test_schema."map_edge_cases"`, `test_schema."interval_edge_cases"`, `test_schema."zonedtimestamp_edge_cases"`, `test_schema."decimal_edge_cases"`}, "id")
```

Comment:
I'd recommend doing a manual verification once (like actually do a select * once and manually verify the outputs ) just because this is some function that we have written (so it could potentially be buggy)